src/components/SelectionList/BaseSelectionList.js

```diff
@@ -0,0 +1,403 @@
+        };
+    }, [canSelectMultiple, sections]);
+
+    const [focusedIndex, setFocusedIndex] = useState(() => {
+        const defaultIndex = 0;
+
+        const indexOfInitiallyFocusedOption = _.findIndex(flattenedSections.allOptions, (option) => option.keyForList === initiallyFocusedOptionKey);
+
+        if (indexOfInitiallyFocusedOption >= 0) {
+            return indexOfInitiallyFocusedOption;
+        }
+
+        return defaultIndex;
+    });
+
+    /**
+     * Scrolls to the desired item index in the section list
+     *
+     * @param {Number} index - the index of the item to scroll to
+     * @param {Boolean} animated - whether to animate the scroll
+     */
+    const scrollToIndex = (index, animated) => {
+        const item = flattenedSections.allOptions[index];
+
+        if (!listRef.current || !item) {
+            return;
+        }
+
+        const itemIndex = item.index;
+        const sectionIndex = item.sectionIndex;
+
+        // Note: react-native's SectionList automatically strips out any empty sections.
+        // So we need to reduce the sectionIndex to remove any empty sections in front of the one we're trying to scroll to.
+        // Otherwise, it will cause an index-out-of-bounds error and crash the app.
+        let adjustedSectionIndex = sectionIndex;
+        for (let i = 0; i < sectionIndex; i++) {
+            if (_.isEmpty(lodashGet(sections, `[${i}].data`))) {
+                adjustedSectionIndex--;
+            }
+        }
+
+        listRef.current.scrollToLocation({sectionIndex: adjustedSectionIndex, itemIndex, animated});
+    };
+
+    const selectRow = (item, index) => {
+        // In single-selection lists we don't care about updating the focused index, because the list is closed after selecting an item
+        if (canSelectMultiple) {
+            if (sections.length === 1) {
+                // If the list has only 1 section (e.g. Workspace Members list), we always focus the next available item
+                const nextAvailableIndex = _.findIndex(flattenedSections.allOptions, (option, i) => i > index && !option.isDisabled);
+                setFocusedIndex(nextAvailableIndex);
+            } else {
+                // If the list has multiple sections (e.g. Workspace Invite list), we focus the first one after all the selected (selected items are always at the top)
+                const selectedOptionsCount = item.isSelected ? flattenedSections.selectedOptions.length - 1 : flattenedSections.selectedOptions.length + 1;
+                setFocusedIndex(selectedOptionsCount);
+            }
+        }
+
+        onSelectRow(item);
+    };
+
+    const selectFocusedOption = () => {
+        const focusedOption = flattenedSections.allOptions[focusedIndex];
+
+        if (!focusedOption || focusedOption.isDisabled) {
+            return;
+        }
+
+        selectRow(focusedOption, focusedIndex);
+    };
```

Comment:
The auto highlight feature is only meant for keyboard navigation right? On click/press we shouldn't highlight any element as this is causing a regression https://github.com/Expensify/App/issues/27377.

We are going to limit that feature to keyboard navigation only on https://github.com/Expensify/App/pull/27646. Let us know if you think we shouldn't.

cc @thiagobrez @Santhosh-Sellavel

Reply:
Hey @s77rt ! This was already fixed here, waiting for a final approval: https://github.com/Expensify/App/pull/27246